src/xpk/core/docker_manager.py

```diff
@@ -95,24 +205,69 @@ def run_command(
       - docker.errors.ImageNotFound,
       - docker.errors.APIError
     """
-    output: bytes = self.client.containers.run(
-        image=img_name,
-        command=cmd,
-        remove=rm_container_after,
-        name=container_name,
-        stdout=True,
-        stderr=True,
-        volumes=[
-            f"{self.gcloud_cfg_path}:{gcloud_cfg_mount_path}",
-            f"{self.deployment_dir}:{deployment_dir_mount_path}",
-        ],
+    xpk_print(f"Running command: {cmd} inside container: {self.container_name}")
+    xpk_print(
+        f"volumes: {self.gcloud_cfg_path}:{gcloud_cfg_mount_path},"
+        f" {self.working_dir}:{working_dir_mount_path}"
     )
-    return output
+    try:
+      self.client.containers.run(
+          image=self.img_name,
+          entrypoint=cmd,
+          remove=self.rm_container_after,
+          name=self.container_name,
+          stdout=True,
+          stderr=True,
+          volumes=[
+              f"{self.gcloud_cfg_path}:{gcloud_cfg_mount_path}",
+              f"{self.working_dir}:{working_dir_mount_path}",
+          ],
+          environment={
+              "GOOGLE_APPLICATION_CREDENTIALS": (
+                  "/root/.config/gcloud/application_default_credentials.json"
+              )
+          },
+      )
+    except ContainerError as e:
+      xpk_print(
+          "Running command failed due to ContainerError with exit status:"
+          f" {e.exit_status} and stderr: {e.stderr}"
+      )
+      xpk_exit(DockerRunCommandExitCode)
+    except ImageNotFound as _:
+      xpk_print(f"Image {ctk_docker_image} not found. Deploying cluster failed")
+      xpk_exit(DockerRunCommandExitCode)
+    except APIError as e:
+      xpk_print(f"Deploying cluster toolkit failed due to {e.explanation}")
+      xpk_exit(DockerRunCommandExitCode)
+
+  def _make_upload_directory(self, name: str) -> str:
```

Comment:
nit: this method is doing two things.  a part of it can be a separate method something like below:
```
  def _get_upload_target_path(self, name: str) -> str:
    target_path = os.path.join(self.working_dir, upload_dir, name)
    return target_path
```